### ISSUE TITLE
Clicking out of command input

### ISSUE THREAD

**Issue by dsande30** [qutebrowser/qutebrowser#3812]
Hello, I'm unsure if this is something anyone else would like implemented but I thought I would mention it.

If you start to enter a command into the browser and then decide to type something into a search bar inside the currently loaded page instead (for example google), when you press enter it executes the typed command instead. Personally I think it would be a tiny (but nice) change to ignore the previously typed command if you start to type in the page instead.

**Comment by toofar** [comment]
I agree that the command mode should go away if you are typing in the page because you should be in insert mode, not command mode. The only reason I can see why you would want to be in command mode and interacting with the page at the same time is if you were crafting some arcane command, but there is `edit-command` for that.

**Comment by NoctuaNivalis** [comment]
I'm interested in the opposite: not allowing to click on the page while you're in command mode. I occasionaly hit my track pad when typing, and when I'm about to type a `d` that's annoying.

Perhaps an option to cancel the prompt/ignore the click would be nice.

**Comment by jgkamat** [comment]
This is probably related to #591 (just to link issues). Solving that should get us the ability to do both of these.

**Comment by The-Compiler** [comment]
I'm not sure whether #591 would help here or not - I don't really have a clear picture in my head yet how mouse bindings would work.

However, I agree the current behavior is confusing. You're still in prompt mode but the website has keyboard focus, so unbound keys get passed there (like in insert mode), but bound keys are still processed in prompt mode.

I think we should do better there, but I do feel like what @NoctuaNivalis proposed is the more reasonable thing to do. I'd like to avoid yet another option, at least if we can get somewhat of a consensus :wink: @dsande30 what do you think about that?

Why I think it's better: If you accidentally click the page while entering a command, you don't lose any input.

Why it might be worse (and leaving command mode would be better): Making the completion usable with the mouse, basically what @HunterGraubard proposed in #2159.

**Comment by dsande30** [comment]
I think that sounds reasonable. It requires more work on the user (having to exit out of prompt mode instead of just clicking), but it prevents mistakes. 

Although maybe clicking links would be allowed? It seems like if I'm clicking a link it wouldn't hurt to just get rid of my typed command altogether and load the new page. That functionality seems tricky though, like what if I want to add a comment on a page and click a submit button. It won't load a new page, but do you allow that?

I have to agree though that any other interaction on the page would be best to just ignore altogether. Perhaps keeping everything simple is the clearest interaction here. Not sure.
